flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/CodeGeneratorContext.scala

```diff
@@ -591,6 +591,18 @@ class CodeGeneratorContext(val tableConfig: TableConfig) {
     fieldTerm
   }
 
+  /**
+    * Adds a reusable boolean LegacyCastBehaviour var to the member ara of the generated class.
+    */
+  def addReusableLegacyCastBehaviour(): Boolean = {
+    val legacyCastBehaviour = tableConfig.getLegacyCastBehaviour
+    val stmt =
+      s"""private static final boolean $DEFAULT_LEGACY_CAST_BEHAVIOUR ="$legacyCastBehaviour";
+      """.stripMargin
+    addReusableMember(stmt)
+    legacyCastBehaviour
+  }
```

Comment:
Why this? Do you expect we need this info it in the codegen?

Reply:
Don't know, I thought to have there prepared just in case, I can remove as well if you think it's better.

Reply:
I personally can't think of any case where we need it, because this configuration is never changed after the planning, and it probably wouldn't make sense to do so, because you could simply break your query, e.g. think about cast that with the new behavior won't return null anymore, it will break the plan as the type of cast becomes `NOT NULL`.

Also, you're not really using this constant anywhere, so I suggest to just remove it.

Reply:
What I think on the other hand would be nice, is to add a comment somewhere in the generated code, perhaps in the class header comment, to log this and other useful info about code generation behavior

Reply:
Makes sense, thx

Reply:
> What I think on the other hand would be nice, is to add a comment somewhere in the generated code, perhaps in the class header comment, to log this and other useful info about code generation behavior

That's a great idea, added in a hotfix commit. Currently the header is added only for the `FunctionCodeGenerator`. If you think we should add this method call in all other code gens already, let me know.